src/custom/state/claim/hooks/index.ts

```diff
@@ -365,3 +366,83 @@ function _hasNoInputOrInputIsGreaterThanClaimAmount(
 ): input is Required<ClaimInput> {
   return !input.amount || JSBI.greaterThan(JSBI.BigInt(input.amount), JSBI.BigInt(claim.amount))
 }
+
+type LastAddress = string
+type ClaimAddressMapping = { [firstAddress: string]: LastAddress }
+const FETCH_CLAIM_MAPPING_PROMISES: Record<number, Promise<ClaimAddressMapping> | null> = {}
+
+/**
+ * Customized fetchClaimMapping function
+ */
+function fetchClaimsMapping(chainId: number): Promise<ClaimAddressMapping> {
+  return (
+    FETCH_CLAIM_MAPPING_PROMISES[chainId] ??
+    (FETCH_CLAIM_MAPPING_PROMISES[chainId] = fetch(`${getClaimsRepoPath(chainId)}mapping.json`)
+      .then((res) => res.json())
+      .catch((error) => {
+        console.error('Failed to get claims mapping', error)
+        FETCH_CLAIM_MAPPING_PROMISES[chainId] = null
+      }))
+  )
+}
+
+const FETCH_CLAIM_FILE_PROMISES: { [startingAddress: string]: Promise<{ [address: string]: RepoClaims }> } = {}
+
+/**
+ * Customized fetchClaimFile function
+ */
+function fetchClaimsFile(address: string, chainId: number): Promise<{ [address: string]: RepoClaims }> {
+  console.log(`fetching key`, address)
```

Comment:
if we want to keep this can you prepend as we usually do: `[Claim:hooks] ... ` or sth similar.

Reply:
Logging statements I forgot to remove. Will clear them out